subprojects/core/src/main/java/org/gradle/testfixtures/internal/TestInMemoryCacheFactory.java

```diff
@@ -149,7 +149,10 @@ public <T> T useCache(Factory<? extends T> action) {
         @Override
         public void useCache(Runnable action) {
```

Comment:
❗ I'm pretty sure not synchronizing here was a bug, which I noticed when my zip file tree unit (not integration) tests suddenly started breaking.

Seems like the `useCache(Runnable action)` overload here would need the same internal synchronization logic as the other overload above it, for the same reason.

I haven't noticed this breaking any other unrelated tests yet, fortunately.